mmv1/products/saasservicemgmt/RolloutKind.yaml

```diff
@@ -0,0 +1,153 @@
+# limitations under the License.
+
+---
+name: RolloutKind
+description: A RolloutKind is a reusable configuration resource that defines the policies, strategies, and targeting for Rollout operations. It acts as a template for repeatable Rollouts, providing guardrails and ensuring that updates are executed in a consistent manner across a fleet of Units.
+base_url: projects/{{project}}/locations/{{location}}/rolloutKinds
+update_mask: true
+self_link: projects/{{project}}/locations/{{location}}/rolloutKinds/{{rollout_kind_id}}
+create_url: projects/{{project}}/locations/{{location}}/rolloutKinds?rolloutKindId={{rollout_kind_id}}
+update_verb: PATCH
+id_format: projects/{{project}}/locations/{{location}}/rolloutKinds/{{rollout_kind_id}}
+import_format:
+  - projects/{{project}}/locations/{{location}}/rolloutKinds/{{rollout_kind_id}}
+min_version: beta
+examples:
+  - name: saas_runtime_rollout_kind_basic
+    primary_resource_id: "example"
+    min_version: "beta"
+    vars:
+      saas_name: example-saas
+      unitkind_name: example-unitkind
+      rolloutkind_name: example-rolloutkind
+    bootstrap_iam:
+      - member: "serviceAccount:service-{project_number}@gcp-sa-saasservicemgmt.iam.gserviceaccount.com"
+        role: "roles/saasservicemgmt.serviceAgent"
+autogen_async: false
+autogen_status: Um9sbG91dEtpbmQ=
+parameters:
+  - name: location
+    type: String
+    description: Resource ID segment making up resource `name`. It identifies the resource within its parent collection as described in https://google.aip.dev/122.
+    immutable: true
+    url_param_only: true
+    required: true
+  - name: rolloutKindId
+    type: String
+    description: The ID value for the new rollout kind.
+    immutable: true
+    url_param_only: true
+    required: true
+properties:
+  - name: annotations
+    type: KeyValueAnnotations
+    description: |-
+      Annotations is an unstructured key-value map stored with a resource that
+      may be set by external tools to store and retrieve arbitrary metadata.
+      They are not queryable and should be preserved when modifying objects.
+
+      More info: https://kubernetes.io/docs/user-guide/annotations
+  - name: createTime
+    type: String
+    description: The timestamp when the resource was created.
+    output: true
+  - name: errorBudget
+    type: NestedObject
+    description: |-
+      The configuration for error budget. If the number of failed units exceeds
+      max(allowed_count, allowed_ratio * total_units), the rollout will be paused.
+    properties:
+      - name: allowedCount
+        type: Integer
+        description: |-
+          The maximum number of failed units allowed in a location without pausing
+          the rollout.
+      - name: allowedPercentage
+        type: Integer
+        description: |-
+          The maximum percentage of units allowed to fail (0, 100] within a location
+          without pausing the rollout.
+  - name: etag
```

Comment:
You can remove this, probably! It should probably be `type: Fingerprint` if not.